src/lib/model/ContentItem.ts

```diff
@@ -173,6 +173,12 @@ export class ContentItem extends HalResource {
         { version: this.version },
         ContentItem
       ),
+
+    /**
+     * Publish content item
+     */
+    publish: (): Promise<void> =>
+      this.performActionWithoutResourceResponse('publish'),
```

Comment:
The `/publish` endpoint returns a `Location:` header to the [publishing-job](https://amplience.com/docs/api/dynamic-content/management/index.html#operation/getPublishingJob) resource that has been created.
To ensure your content has been published, you must check the `state` property on the publishing-job.

As a consumer of the SDK I would want to know when the publish has finished, so either this function can return a `PublishingJob` resource along with a method that I can call to wait until the publishing-job has finished or for the `publish()` method to await for the job to be in a completed state.

```
const contentItem = await client.contentItems.get('<CONTENT-ITEM-ID>');
const publishingJob = await contentItem.related.publish()
await publishingJob.waitUntilFinished()
```

Or...
```
const contentItem = await client.contentItems.get('<CONTENT-ITEM-ID>');
const publishingJob = await contentItem.related.publish(); // waits until the publishing job has finished
```